include/bluetooth/services/hids_c.h

```diff
@@ -243,7 +206,25 @@ struct bt_gatt_hids_c_init_params {
 };
 
 /**
- * HIDS client structure
+ * @brief Report base information struct
+ *
+ * The structure that contains record information.
+ * This is common structure for Input and Output/Feature reports.
+ * Some of the fields are used only for Input report type.
+ *
+ */
+struct bt_gatt_hids_c_rep_info;
+
+/**
+ * @brief HIDS client structure
+ *
+ * This structure is used ad HIDS client object.
+ *
+ * @note
+ * It is defined here to allow the user to allocate the memory for it
```

Comment:
maybe we could do the following rename the structure to `struct bt_gatt_hids_c_private` and do
```
struct bt_gatt_hids_c {
   bt_gatt_hids_c_priv priv; // no touching
}
```
performance wise it would be the same but reference to the o->priv.x will be needed everywhere so I am not very sure this is the good idea...

Reply:
We can go this way. Not quite sure about it as here whole structure should not be accessed by the user. If more people would vote for it I will update it.